org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterColorColor.java

```diff
@@ -443,36 +441,26 @@ public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
 
     private void updateOnOff(OnOffType onOff) {
         boolean on = onOff == OnOffType.ON;
-        currentOnOffState.set(on);
-
-        if (lastColorMode != ColorModeEnum.COLOR_TEMPERATURE) {
-            // Extra temp variable to avoid thread sync concurrency issues on lastHSB
-            HSBType oldHSB = lastHSB;
-            HSBType newHSB = on ? lastHSB : new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), PercentType.ZERO);
-            updateChannelState(newHSB);
-        } else if (!on) {
-            updateChannelState(OnOffType.OFF);
-        }
+        // Extra temp variable to avoid thread sync concurrency issues on lastHSB
+        HSBType oldHSB = lastHSB;
+        HSBType newHSB = on ? lastHSB : new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), PercentType.ZERO);
+        updateChannelState(newHSB);
     }
 
     private void updateBrightness(PercentType brightness) {
         // Extra temp variable to avoid thread sync concurrency issues on lastHSB
         HSBType oldHSB = lastHSB;
         HSBType newHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), brightness);
         lastHSB = newHSB;
-        if (currentOnOffState.get() && lastColorMode != ColorModeEnum.COLOR_TEMPERATURE) {
-            updateChannelState(newHSB);
-        }
+        updateChannelState(newHSB);
     }
 
     private void updateColorHSB(DecimalType hue, PercentType saturation) {
         // Extra temp variable to avoid thread sync concurrency issues on lastHSB
         HSBType oldHSB = lastHSB;
         HSBType newHSB = new HSBType(hue, saturation, oldHSB.getBrightness());
         lastHSB = newHSB;
-        if (currentOnOffState.get() && lastColorMode != ColorModeEnum.COLOR_TEMPERATURE) {
```

Comment:
I think this introduces a bug.  This removed the check of `currentOnOffState` and just calls `updateChannelState`. Therefore the channel can be updated even if the device is off, and we end up with the incorrect state being propagated.

Reply:
@andrewfg I see you completely removed the management of the `currentOnOffState` - before I put it back again, was there a reason for this, or was it just that you weren't aware of it's purpose?

Reply:
I will look at it tomorrow. But in the meantime my off the cuff answer is that the HS parts of HSB are orthogonal to the B part thereof. So the B part will reflect changes in Brightness and Onoff state, whereas the HS part would reflect changes in actual hue and saturation.

Reply:
If the device is off, then B should always be 0 - right? At the moment, you removed the handling of the on/off so the B is always reflected to the user even if the device is off, which means it now shows it's on.

Reply:
Note there's a PR in #889 to put these checks back again.